numpyro/infer/svi.py

```diff
@@ -116,6 +116,9 @@ def init(self, rng_key, *args, **kwargs):
                 params[site['name']] = transform.inv(site['value'])
 
         self.constrain_fn = partial(transform_fn, inv_transforms)
+        # we convert weak types like float to float32/float64
+        # to avoid recompiling body_fn in svi.run
+        params = tree_map(lambda x: lax.convert_element_type(x, jnp.result_type(x)), params)
```

Comment:
@martinjankowiak Your observation is right: compiling time is slower in the recent jax release due to this issue.